vm/src/system.rs

```diff
@@ -645,6 +645,7 @@ where
                 contract_addr,
                 admin: new_admin,
             } => {
+                vm.addr_validate(&new_admin)??;
```

Comment:
Did we check the reference implementation for this?

Reply:
I forgot to push my latest commit actually. We don't need this at all since `new_admin.try_into()?` passes the address checking to the VM implementer. At that point, it is up to the VM implementer to check the address generate the address from a string. So we are good.

Reply:
:+1: